maven-wrapper-distribution/src/resources/only-mvnw.cmd

```diff
@@ -85,6 +97,293 @@ if ($env:MAVEN_USER_HOME) {
+
+          if ($matchingVersion) {
+            $normalizedVersion = $matchingVersion -replace [regex]::Escape($sdkmanSuffix), ''
+
+            # Cache the result (unless policy is 'always')
+            if ($updatePolicy -ne "always") {
+              if (-not (Test-Path -Path $cacheDir)) {
+                New-Item -ItemType Directory -Path $cacheDir -Force | Out-Null
+              }
+              Set-Content -Path $cacheFile -Value $normalizedVersion -ErrorAction SilentlyContinue
+            }
+
+            Write-Verbose "Resolved JDK version from SDKMAN (policy: $updatePolicy): $Version -> $normalizedVersion"
+        } else {
+          # Show helpful error message with available alternatives
+          $vendorName = switch ($sdkmanSuffix) {
+            "-tem" { "temurin" }
+            "-amzn" { "corretto" }
+            "-zulu" { "zulu" }
+            "-librca" { "liberica" }
+            "-oracle" { "oracle" }
+            "-grl" { "graalvm" }
+            "-ms" { "microsoft" }
+            "-sem" { "semeru" }
+            default { "unknown" }
+          }
+
+          $availableVersions = $versions | Where-Object { $_ -like "$Version.*" } | ForEach-Object { ($_ -split '-')[-1] } | Sort-Object -Unique
+
+          Write-Error "ERROR: JDK $Version is not available from vendor '$vendorName'."
+          Write-Error ""
+          if ($availableVersions) {
+            Write-Error "Available JDK $Version vendors:"
+            $availableVersions | ForEach-Object { Write-Error "  - $_" }
+            Write-Error ""
+            Write-Error "To use a different vendor, set jdkVendor in maven-wrapper.properties:"
+            Write-Error "  jdkVendor=temurin"
+            Write-Error "  jdkVendor=oracle"
+            Write-Error "  jdkVendor=corretto"
+          } else {
+            Write-Error "JDK $Version is not available from any vendor."
+            $majorVersions = $versions | ForEach-Object { ($_ -split '\.')[0] } | Sort-Object -Unique | Select-Object -First 10
+            Write-Error ""
+            Write-Error "Available JDK major versions:"
+            $majorVersions | ForEach-Object { Write-Error "  - $_" }
+          }
+          Write-Error ""
+            Write-Error "Alternatively, specify an exact JDK URL with jdkDistributionUrl."
+            return
+          }
+        } catch {
+          # Network or API error
+          Write-Error "Failed to resolve JDK version $Version from SDKMAN API: $($_.Exception.Message)"
+          Write-Error "Please check your internet connection or specify an exact JDK version with jdkDistributionUrl."
+          return
+        }
+      }
+    }
+
+    $sdkmanVersion = "$normalizedVersion$sdkmanSuffix"
+    $platform = "windowsx64"  # Windows x64 platform identifier for SDKMAN
+
+    # Call SDKMAN API to get download URL (handle 302 redirect)
+    $sdkmanApiUrl = "https://api.sdkman.io/2/broker/download/java/$sdkmanVersion/$platform"
```

Comment:
let's warn this will be unstable compare to passing an url

for known vendor we should switch to stable urls (zulu, temurin etc) probably